src/index/hgraph_zparameters.cpp

```diff
@@ -0,0 +1,101 @@
+
+// Copyright 2024-present the vsag project
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "hgraph_zparameters.h"
+
+#include "../utils.h"
+#include "fmt/format-inl.h"
+#include "inner_string_params.h"
+#include "vsag/constants.h"
+
+namespace vsag {
+
+HgraphParameters::HgraphParameters(const IndexCommonParam& common_param, const std::string& str)
+    : common_param_(common_param) {
+    this->sync_json_by_string();
+    this->ParseStringParam(str);
+}
+
+void
+HgraphParameters::ParseStringParam(const std::string& str) {
+    auto json_obj = nlohmann::json::parse(str);
+    for (const auto& [key, value] : json_obj.items()) {
+        this->CheckAndSetKeyValue(key, value);
+    }
+}
+
+void
+HgraphParameters::CheckAndSetKeyValue(const std::string& key, nlohmann::json& value) {
```

Comment:
nlohmann::json can define to `JsonType` ? like https://github.com/alipay/vsag/pull/96

Reply:
we will alias all of this json type in another PR

Reply:
OK